bin/stacks/routing-caching-stack.ts

```diff
@@ -72,40 +79,85 @@ export class RoutingCachingStack extends cdk.NestedStack {
-        POOL_CACHE_BUCKET_2: this.poolCacheBucket2.bucketName,
-        POOL_CACHE_KEY: this.poolCacheKey,
-      },
-    })
-
-    this.poolCacheBucket.grantReadWrite(this.poolCacheLambda)
-    this.poolCacheBucket2.grantReadWrite(this.poolCacheLambda)
+    const lambdaLayerVersion = aws_lambda.LayerVersion.fromLayerVersionArn(
+      this,
+      'InsightsLayerPools',
+      `arn:aws:lambda:${region}:580247275435:layer:LambdaInsightsExtension:14`
+    )
 
-    new aws_events.Rule(this, 'SchedulePoolCache', {
-      schedule: aws_events.Schedule.rate(Duration.minutes(15)),
-      targets: [new aws_events_targets.LambdaFunction(this.poolCacheLambda)],
-    })
+    // Spin up a new pool cache lambda for each config in chain X protocol
+    for (let i = 0; i < chainProtocols.length; i++) {
+      const { protocol, chainId, timeout } = chainProtocols[i]
+      const lambda = new aws_lambda_nodejs.NodejsFunction(
+        this,
+        `PoolCacheLambda-ChainId${chainId}-Protocol${protocol}`,
+        {
+          role: lambdaRole,
+          runtime: aws_lambda.Runtime.NODEJS_14_X,
+          entry: path.join(__dirname, '../../lib/cron/cache-pools.ts'),
+          handler: 'handler',
+          timeout: Duration.seconds(900),
+          memorySize: 1024,
+          bundling: {
+            minify: true,
+            sourceMap: true,
+          },
+          description: `Pool Cache Lambda for Chain with ChainId ${chainId} and Protocol ${protocol}`,
+          layers: [lambdaLayerVersion],
+          tracing: aws_lambda.Tracing.ACTIVE,
+          environment: {
+            POOL_CACHE_BUCKET: this.poolCacheBucket.bucketName,
+            POOL_CACHE_BUCKET_2: this.poolCacheBucket2.bucketName,
+            POOL_CACHE_KEY: this.poolCacheKey,
+            chainId: chainId.toString(),
+            protocol,
+            timeout: timeout.toString(),
+          },
+        }
+      )
+      new aws_events.Rule(this, `SchedulePoolCache-ChainId${chainId}-Protocol${protocol}`, {
+        schedule: aws_events.Schedule.rate(Duration.minutes(15)),
+        targets: [new aws_events_targets.LambdaFunction(lambda)],
+      })
+      this.poolCacheBucket2.grantReadWrite(lambda)
+      const lambdaAlarmErrorRate = new aws_cloudwatch.Alarm(
+        this,
+        `RoutingAPI-PoolCacheToS3LambdaErrorRate-ChainId${chainId}-Protocol${protocol}`,
+        {
+          metric: new MathExpression({
+            expression: '100*(errors/invocations)',
+            usingMetrics: {
+              invocations: lambda.metricInvocations({
+                period: Duration.minutes(60),
+                statistic: 'sum',
+              }),
+              errors: lambda.metricErrors({
+                period: Duration.minutes(60),
+                statistic: 'sum',
+              }),
+            },
+          }),
+          threshold: protocol === Protocol.V3 ? 50 : 85,
+          evaluationPeriods: protocol === Protocol.V3 ? 12 : 60,
```

Comment:
Are you sure this is set correctly? My understand is that evaluation periods is set based on the metric field, so since we specified period as "60 minutes" above, then this would be 12 * 60 minutes so 12 hours no?

Reply:
the evaluation period is 5 minutes, so its 12 * 5 minutes = 1 hour